components/Markets/ConditionalMarketOrderBook.tsx

```diff
@@ -10,107 +17,288 @@ export function ConditionalMarketOrderBook({
   orderBookObject: _OrderBook;
   setPriceFromOrderBook: (price: string) => void;
 }) {
-  let bids = orderBookObject?.failBidsArray;
-  let asks = orderBookObject?.failAsksArray;
-  let spreadString = orderBookObject?.failSpreadString;
+  if (!orderBookObject) return null;
+  const { colorScheme } = useMantineColorScheme();
+  const provider = useProvider();
+  const proposal = useProposal();
+  const [bids, setBids] = useState<any[][]>();
+  const [asks, setAsks] = useState<any[][]>();
+  const [wsConnected, setWsConnected] = useState<boolean>(false);
+  const [spreadString, setSpreadString] = useState<string>();
+  const [lastSlotUpdated, setLastSlotUpdated] = useState<number>(0);
+  const openBookProgram = new Program<OpenbookV2>(OPENBOOK_IDL, OPENBOOK_PROGRAM_ID, provider);
+
+  // On initialization
   if (isPassMarket) {
-    bids = orderBookObject?.passBidsArray;
-    asks = orderBookObject?.passAsksArray;
-    spreadString = orderBookObject?.passSpreadString;
+    if (!bids) {
+      setBids(orderBookObject.passBidsArray);
+    }
+    if (!asks) {
+      setAsks(orderBookObject.passAsksArray);
+    }
+    if (!spreadString) {
+      setSpreadString(orderBookObject.passSpreadString);
+    }
+  } else {
+    if (!bids) {
+      setBids(orderBookObject.failBidsArray);
+    }
+    if (!asks) {
+      setAsks(orderBookObject.failAsksArray);
+    }
+    if (!spreadString) {
+      setSpreadString(orderBookObject.failSpreadString);
+    }
   }
 
-  return (
-    <>
-      <style
-        dangerouslySetInnerHTML={{
-          __html: `
-        .MakeItNice {
-          font-family: -apple-system, BlinkMacSystemFont, sans-serif;
-          font-size: 15px;
-          font-variant-numeric: tabular-nums;
-          width: 100%;
-          display: inline-block;
-        }
-        .MakeItNice__side--bids, .MakeItNice__side--asks {
-          min-height:130px;
-          width: 100%;
-          display: flex;
-        }
+  const consumeOrderBookSide = (
+    side: string,
+    updatedAccountInfo: AccountInfo<Buffer>,
+    ctx: Context
+  ) => {
+    try {
+      const leafNodes = openBookProgram.coder.accounts.decode('bookSide', updatedAccountInfo.data);
+      const leafNodesData = leafNodes.nodes.nodes.filter(
+        (x: AnyNode) => x.tag === 2,
```

Comment:
Can you add a comment here? What's `2`? Or perhaps besides a comment, create a descriptive constant and set it = to 2

Reply:
Links to relevant docs in the comments would help